tests/mocha/contextmenu_items_test.js

```diff
@@ -0,0 +1,430 @@
+        chai.assert.equal(workspace.getTopBlocks(false).length, 0);
+      });
+
+      test('Has correct label for multiple blocks', function() {
+        workspace.newBlock('text');
+        workspace.newBlock('text');
+
+        chai.assert.equal(deleteOption.displayText(scope), 'Delete 2 Blocks');
+      });
+
+      test('Has correct label for single block', function() {
+        workspace.newBlock('text');
+        chai.assert.equal(deleteOption.displayText(scope), 'Delete Block');
+      });
+    });
+  });
+
+  suite('Block Items', function() {
+    var scope;
+    var block;
+
+    setup(function() {
+      block = workspace.newBlock('text');
+      scope = {block: block};
+    });
+
+    teardown(function() {
+      block.dispose();
+    });
+
+    suite('Duplicate', function() {
+      var duplicateOption;
+      
+      setup(function() {
+        duplicateOption = registry.getItem('blockDuplicate');
+      });
+
+      test('Enabled when block is duplicatable', function() {
+        // Block is duplicatable by default
+        chai.assert.equal(duplicateOption.preconditionFn(scope), 'enabled');
+      });
+
+      test('Disabled when block is not dupicatable', function() {
+        sinon.stub(block, 'isDuplicatable').returns(false);
+        chai.assert.equal(duplicateOption.preconditionFn(scope), 'disabled');
+      });
+
+      test('Hidden when in flyout', function() {
+        block.isInFlyout = true;
+        chai.assert.equal(duplicateOption.preconditionFn(scope), 'hidden');
+      });
+
+      test('Calls duplicate', function() {
+        var stub = sinon.stub(Blockly, 'duplicate');
+
+        duplicateOption.callback(scope);
+
+        sinon.assert.calledOnce(stub);
+        sinon.assert.calledWith(stub, block);
+      });
+
+      test('Has correct label', function() {
+        chai.assert.equal(duplicateOption.displayText(), 'Duplicate');
+      });
+    });
+
+    suite('Comment', function() {
+      var commentOption;
+
+      setup(function() {
+        commentOption = registry.getItem('blockComment');
+      });
+
+      test('Enabled for normal block', function() {
+        chai.assert.equal(commentOption.preconditionFn(scope), 'enabled');
+      });
+
+      test('Hidden for IE', function() {
+        Blockly.utils.userAgent.IE = true;
+        chai.assert.equal(commentOption.preconditionFn(scope), 'hidden');
+      });
+
+      test('Hidden for collapsed block', function() {
+        // Must render block to collapse it properly.
+        block.initSvg();
+        block.render();
+        block.setCollapsed(true);
+
+        chai.assert.equal(commentOption.preconditionFn(scope), 'hidden');
+      });
+
+      test('Creates comment if one did not exist', function() {
+        chai.assert.isNull(block.getCommentIcon(), 'New block should not have a comment');
+        commentOption.callback(scope);
+        chai.assert.isEmpty(block.getCommentText(), 'Block should have empty comment text');
```

Comment:
It seems like maybe you should also add a `chai.assert.exists(block.getCommentIcon())`

Reply:
Done